source/sync/migrate-sync-modes.txt

```diff
@@ -0,0 +1,192 @@
+clients communicate with a Partition-Based Syn App Services App.
+
+Requirements
+------------
+
+- Your App must currently use Partition-Based Sync
+- Linked Atlas cluster must be running mongodb 5.0+ for Flexible Sync compatibility
+- Linked Atlas cluster must have time-based oplog setting enabled
+- realm-core vTBD+, which means these SDK versions: TBD
+
+.. _realm-sync-migrate-what-to-know:
+
+What to Know Before migrating
+-----------------------------
+
+- How long a migration takes is directly proportional to the amount of data
+  that needs to be migrated.
+- Migrations can be canceled or rolled back within a certain timeframe.
+- As part of migration, sync data is doubled. This may be enough to impact
+  billing.
+- Progress notifications will not work after migration because Flexible Sync does
+  note yet support them.
+
+.. _realm-sync-migrate-stages:
+
+Migration Stages
+~~~~~~~~~~~~~~~~
+
+#. Flushing partition-based sync
+   
+   Clients can connect at this stage, but local writes will not sync.
+	 Clients can receive updates to MongoDB data.
+
+#. Initializing flexible sync
+   
+   Clients cannot connect at this stage. The time this takes is directly
+   proporional to the amount of data in synced collections.
+
+#. Migration completed
+
+   **Manage sync migration** appears on Sync page if the migration hasn't been 
+   committed or rolled back. This button lets admins commit or roll back manually.
+
+Migrate Partition-Based Sync App to Flexible Sync
+-------------------------------------------------
+
+After beginning migration from Partition-Based Sync to Flexible Sync, the process
+is mostly automatic. However, you can control important parts of the migration.
+
+During a migration, you can cancel it. After a migration, you can roll it back 
+or manually commit the migration.
+
+Begin Migration
+~~~~~~~~~~~~~~~
+
+Before you begin a migration, make sure you're aware of the
+:ref:`migration stages <realm-sync-migrate-stages>` and
+:ref:`effects <realm-sync-migrate-what-to-know>`.
+
+During migration, client apps may experience side effects, most of which are similar
+to what happens if you disable and re-enable sync. These are some side effects
+of migration:
+
+- In the App Services UI, the Rules and Schemas pages will be unresponsive.
+- Your App Services Appp will experience a period of read-only sync and then a 
+  period of downtime.
+- If Partition-Based Sync default client max offline time is unset, the new
+  Flexible Sync config will set it to 30 days.
+- Storage in your Atlas cluster used by sync is temporarily doubled. Partition-Based
+  Sync and Flexible Sync must temporarily co-exist. Depending on the size of your
+  synced data, this could affect billing.
+
+.. QUESTION: If someone were to cancel a migration - or roll back a migration,
+   they'd potentially be on the hook for costs associated with subsequent 
+   migration attempts, correct?
```

Comment:
what costs are you referring to here?

Reply:
Ah, sorry. Potential costs related to the doubling sync metadata. It's doubled every time because they have to start over, right?

Reply:
Ah I see. Canceling/rolling back a migration automatically clears the flexible sync metadata - so their storage should go back to what it was prior to starting the migration